egs2/TEMPLATE/vad1/vad.sh

```diff
@@ -0,0 +1,947 @@
+            _dir="${vad_exp}/${inference_tag}/${dset}"
+            _logdir="${_dir}/logdir"
+            mkdir -p "${_logdir}"
+
+            _feats_type="$(<${_data}/feats_type)"
+            if [ "${_feats_type}" = raw ]; then
+                _scp=wav.scp
+                if [[ "${audio_format}" == *ark* ]]; then
+                    _type=kaldi_ark
+                else
+                    _type=sound
+                fi
+            else
+                _scp=feats.scp
+                _type=kaldi_ark
+            fi
+
+            # 1. Split the key file
+            key_file=${_data}/${_scp}
+            split_scps=""
+
+            _nj=$(min "${inference_nj}" "$(<${key_file} wc -l)")
+
+            for n in $(seq "${_nj}"); do
+                split_scps+=" ${_logdir}/keys.${n}.scp"
+            done
+            # shellcheck disable=SC2086
+            utils/split_scp.pl "${key_file}" ${split_scps}
+
+            # 2. Submit decoding jobs
+            log "Decoding started... log: '${_logdir}/vad_inference.*.log'"
+            rm -f "${_logdir}/*.log"
+            # shellcheck disable=SC2046,SC2086
+            ${_cmd} --gpu "${_ngpu}" JOB=1:"${_nj}" "${_logdir}"/vad_inference.JOB.log \
+                ${python} -m espnet2.bin.vad_inference \
+                    --batch_size ${batch_size} \
+                    --ngpu "${_ngpu}" \
+                    --data_path_and_name_and_type "${_data}/${_scp},speech,${_type}" \
+                    --key_file "${_logdir}"/keys.JOB.scp \
+                    --vad_train_config "${vad_exp}"/config.yaml \
+                    --vad_model_file "${vad_exp}"/"${inference_vad_model}" \
+                    --output_dir "${_logdir}"/output.JOB \
+                    ${_opts} ${inference_args} || { cat $(grep -l -i error "${_logdir}"/vad_inference.*.log) ; exit 1; }
+
+            # 3. Concatenates the output files from each jobs
+            # shellcheck disable=SC2068
+            for ref_txt in ${ref_text_files[@]}; do
+                suffix=$(echo ${ref_txt} | sed 's/text//')
+                if [ -f "${_logdir}/output.1/vad_result/segments" ]; then
+                    for i in $(seq "${_nj}"); do
+                        cat "${_logdir}/output.${i}/vad_result/segments"
+                    done | sort -k1 >"${_dir}/segments"
+                fi
+            done
+
+        done
+    fi
+
+
+    if [ ${stage} -le 7 ] && [ ${stop_stage} -ge 7 ]; then
+        log "Stage 7: Scoring"
+
+        for dset in ${test_sets}; do
+            _data="${data_feats}/${dset}"
+            _dir="${vad_exp}/${inference_tag}/${dset}"
+            _logdir="${_dir}/logdir"
+
+            # shellcheck disable=SC2068
+            for ref_txt in ${ref_text_files[@]}; do
+                # Note(simpleoier): to get the suffix after text, e.g. "text_spk1" -> "_spk1"
+                suffix=$(echo ${ref_txt} | sed 's/text//')
+
+            done
+
+            ${python} -m espnet2.bin.vad_scoring \
```

Comment:
Usually, for scoring, we would prefer putting them to pyscripts or scripts instead of binary